pkg/client/restclient/plugin.go

```diff
@@ -30,9 +30,16 @@ type AuthProvider interface {
 	// WrapTransport allows the plugin to create a modified RoundTripper that
 	// attaches authorization headers (or other info) to requests.
 	WrapTransport(http.RoundTripper) http.RoundTripper
+	// Login is expected to maybe require user interaction; it won't be called
+	// in environments where a user is not present
+	Login() error
 }
 
-type Factory func() (AuthProvider, error)
+type Factory func(string, map[string]string, AuthProviderConfigPersister) (AuthProvider, error)
+
```

Comment:
Please also document what these args are for - especially the super generic ones - the string and the map.